### PR TITLE
90 implement a 404 page for unauthorized routes

### PR DESCRIPTION
**Pull request by jejoivanic**
## Feature
Added redirection to unauthorize view depending on the user assigned permissions and the view required permissions the user is trying to access.
The unauthorized component should be passed as a property to the Admin component. If none is provided then a default view is shown with a default message.
Also a (first aproach) Unauthorized Core component was created for helping the user write less code in the template section
## How to use
You just have to create a component to display the information you want to display in the unauthorized page and pass it to the Admin component like:
```
<Admin :authProvider="authProvider" :unauthorized="UnauthorizedCustomView">
```
## The Unauthorized component
```
  <Unauthorized>
    <h2 slot="title">You do not have permissions for accessing this view</h2>
    <div slot="text">
      <p>If you think you should be allowed to see this view, please contact the page administrator.</p>
      <v-btn class="m-0" color="success" v-on:click="goBack">
          BACK
      </v-btn>
    </div>
  </Unauthorized>
```
As you see, we have two "sections", the one with `<h2 slot="title">` will be part of the title in the upper part of the rendered view and the `<div slot="text">` will be part of the text in the bottom part of the rendered view.
This component will surely be improved.

### CODE REVIEW

File: src/router/route.hooks.js [hunk 1]
```diff
@@ -42,9 +42,7 @@ export default({
           } else {
             // User is authenticated but does not have route permissions
             // TODO: Should redirect to an Anauthorized page - #90 - @sgobotta
```
Comment (sgobotta) [inline src/router/route.hooks.js]: This comment can surely be removed now =]

File: tests/e2e/specs/unauthorized/unauthorized.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+describe('Unauthorized User Test', () => {
+  const testResourceName = 'articles'
+  const unauthorizedPath = 'unauthorized'
+  const userWithoutPermissions = Factory.createUser({ permissions: []})
+  const userWithPermissions    = Factory.createUser({ permissions: ['admin']})
+  const createAuthResponse = (user) => Factory.createAuthResponse({ user })
+  const authResponseWithUserWithoutPermissions = createAuthResponse(userWithoutPermissions)
+  const authResponseWithUserWithPermissions = createAuthResponse(userWithPermissions)
+  
+  it('Not admin user should be redirected when trying to access unauthorized view', () => {
```
Comment (sgobotta) [inline tests/e2e/specs/unauthorized/unauthorized.spec.js]: I'd suggest to rename this test a lil bit to: *User without admin permissions should be... access an unauthorized...*. Do you think it fits?

File: tests/e2e/specs/unauthorized/unauthorized.spec.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    cy.InitAuthenticatedUser({ authResponse: authResponseWithUserWithoutPermissions })
+    cy.visit(`/#/${testResourceName}/`)
+    cy.url().should('include', `/#/${unauthorizedPath}`)
+  })
+
+  it('Admin user should not be redirected when trying to access unauthorized view', () => {
```
Comment (sgobotta) [inline tests/e2e/specs/unauthorized/unauthorized.spec.js]: I'd do the same here: *A user with admin privileges... an unauthorized...*

File: tests/e2e/specs/unauthorized/unauthorized.spec.js (hunk 2 above)
Comment (jejoivanic) [reply tests/e2e/specs/unauthorized/unauthorized.spec.js]: Totally agree!

**Pull request merged**